test_agent/rust/src/utils.rs

```diff
@@ -0,0 +1,528 @@
+    let name = if let Some(authority_value) = value
+        .get("authority")
+        .and_then(|authority_value| authority_value.get("name"))
+        .and_then(|authority_value| authority_value.as_str())
+    {
+        Some(authority_value.to_owned())
+    } else {
+        return Err(serde::de::Error::custom("Missing value authority name"));
+    };
+
+    let number = if let Some(authority_number_ip) = value
+        .get("authority")
+        .and_then(|authority_value| authority_value.get("number"))
+        .and_then(|number| number.get("ip"))
+    {
+        Some(up_rust::Number::Ip(
+            authority_number_ip.to_string().as_bytes().to_vec(),
+        ))
+    } else if let Some(authority_number_id) = value
+        .get("authority")
+        .and_then(|authority_value| authority_value.get("number"))
+        .and_then(|number| number.get("id"))
+    {
+        Some(up_rust::Number::Id(
+            authority_number_id.to_string().as_bytes().to_vec(),
+        ))
+    } else {
+        return Err(serde::de::Error::custom("Missing value number"));
+    };
+
+    Ok(UAuthority {
+        name,
+        number,
+        special_fields: SpecialFields::default(),
+    })
+}
+#[derive(Default)]
+pub struct WrapperUAttribute(pub UAttributes);
+impl<'de> Deserialize<'de> for WrapperUAttribute {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let value: Value = Deserialize::deserialize(deserializer)?;
+        let mut uattributes = UAttributes::new();
+
+        if let Some(priority_value) = value.get("priority").and_then(|v| v.as_str()) {
+            uattributes.priority = UPriority::from_str(priority_value)
+                .unwrap_or_else(|| {
+                    // Handle the case where the conversion fails
+                    error!("Error:: Something wrong with priority field");
+                    UPriority::UPRIORITY_UNSPECIFIED
+                })
+                .into();
+        } else {
+            error!("Error:pririty value is not string!");
+        }
+
+        debug!("uattributes.priority: {:?}", uattributes.priority);
+
+        if let Some(type_value) = value.get("type").and_then(|v| v.as_str()) {
+            uattributes.type_ = UMessageType::from_str(type_value)
+                .unwrap_or_else(|| {
+                    // Handle the case where the conversion fails
+                    error!("Error: Something wrong with type field");
+                    UMessageType::UMESSAGE_TYPE_UNSPECIFIED
+                })
+                .into();
+        } else {
+            error!("Error: type value is not string!");
+        }
+
+        debug!("uattributes.type_: {:?}", uattributes.type_);
+
+        if let Some(source_value) = value.get("source") {
+            if let Ok(wrapper_uri) = serde_json::from_value::<WrapperUUri>(source_value.clone()) {
+                uattributes.source = MessageField(Some(Box::new(wrapper_uri.0)));
```

Comment:
Prefer using the Into trait (which will handle the conversion for you) to reduce line noise:
```suggestion
                uattributes.source = Some(wrapper_uri.0).into();
```
Same goes for the rest of the same sorts of lines.

Reply:
done, except for line ~350, where the same sort of fix doesn't work for reqid, will mark after commit